app/models/user.rb

```diff
@@ -0,0 +1,15 @@
+class User < ApplicationRecord
+  has_one_attached :profile_picture
+
+  EMAIL_REGEX = /\A[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,4}\z/i
+  enum :role, { admin: 0, teacher: 1, student: 2 }
+  validates :name, presence: true
+  validates :email, presence: true, length: { in: 10..50 }, format: { with: EMAIL_REGEX }, uniqueness: true
+  validates :role, presence: true, inclusion: { in: roles.keys }
+
+  has_many :classrooms, foreign_key: :user_id, dependent: :destroy
+  has_many :posts, dependent: :destroy
+  has_many :comments, dependent: :destroy
+  has_many :enrollments, dependent: :destroy
+  has_many :classrooms_as_student, through: :enrollments, source: :classroom
```

Comment:
What about following associations: 
User(Teacher) should have many students and User(Student) should have many teachers?

Reply:
Bhai instead of having this association, I have the association that a classroom has many students. And students are enrolled in a classroom. So the teacher can see the students which are enrolled in the classroom they created.